### PR TITLE
[management] When collecting group and peer IDs from policies, do so directionally

### PR DESCRIPTION
**Pull request by dmitri-netbird**
## Describe your changes
In affected peer calculation, when walking policies of groups and peers, only use opposite sides of those policies; i.e. if a policy rule sources include the group, use the rule's destinations, and vice-versa. Similarly, if a rule's SourceResource matches a peer from the change set, use rule's DesstinationResource and vice-versa.

### Checklist
- [ ] Is it a bug fix
- [ ] Is a typo/documentation fix
- [x] Is a feature enhancement
- [ ] It is a refactor
- [ ] Created tests that fail without the change (if possible)
- [x] This change does **not** modify the public API, gRPC protocols, functionality behavior, CLI / service flags, or introduce a new feature — **OR** I have discussed it with the NetBird team beforehand (link the issue / Slack thread in the description). See [CONTRIBUTING.md](https://github.com/netbirdio/netbird/blob/main/CONTRIBUTING.md#discuss-changes-with-the-netbird-team-first).

> By submitting this pull request, you confirm that you have read and agree to the terms of the [Contributor License Agreement](https://github.com/netbirdio/netbird/blob/main/CONTRIBUTOR_LICENSE_AGREEMENT.md).

## Documentation
Select exactly one:

- [ ] I added/updated documentation for this change
- [x] Documentation is **not needed** for this change (explain why)

### Docs PR URL (required if "docs added" is checked)
Paste the PR link from https://github.com/netbirdio/docs here:

https://github.com/netbirdio/docs/pull/__

## Summary by CodeRabbit

* **Refactor**
  * Improved how policy changes are analyzed to determine the affected peer and group impact, using directionally-derived extraction from policy rules for more accurate and deterministic results.

* **Tests**
  * Reworked resolver unit tests to validate precise affected peer/group sets derived from policy source/destination direction rules, including expanded coverage for mixed resource types and allowlist edge cases.
  * Updated affected-peers coverage-matrix expectations and tightened assertions to compare exact included/excluded sets.

### CODE REVIEW

File: management/server/affectedpeers/resolver.go [hunk 1]
```diff
@@ -734,22 +742,40 @@ func collectPolicySources(policy *types.Policy, groupSet, peerSet map[string]str
...
+	groupIds := make([]string, 0)
 	for _, rule := range policy.Rules {
-		if anyInSet(rule.Sources, groupSet) || anyInSet(rule.Destinations, groupSet) {
-			return true
+		// TODO (dmitri) can a group to be present on both sides of a policy?
+		if anyInSet(rule.Sources, groupSet) {
+			groupIds = append(groupIds, rule.Destinations...)
+		} else if anyInSet(rule.Destinations, groupSet) {
+			groupIds = append(groupIds, rule.Sources...)
 		}
```
Comment (pascal-fischer) [inline management/server/affectedpeers/resolver.go]: Yes a group can be present on both sides

File: management/server/affectedpeers/resolver.go [hunk 2]
```diff
@@ -734,22 +742,40 @@ func collectPolicySources(policy *types.Policy, groupSet, peerSet map[string]str
...
+		// TODO (dmitri) can a peer to be present on both sides of a policy?
+		if isDirectPeerInSet(rule.SourceResource, changedSet) {
+			if rule.DestinationResource.Type == types.ResourceTypePeer && rule.DestinationResource.ID != "" {
+				peerIds = append(peerIds, rule.DestinationResource.ID)
+			}
+		} else if isDirectPeerInSet(rule.DestinationResource, changedSet) {
+			if rule.SourceResource.Type == types.ResourceTypePeer && rule.SourceResource.ID != "" {
+				peerIds = append(peerIds, rule.SourceResource.ID)
+			}
 		}
```
Comment (pascal-fischer) [inline management/server/affectedpeers/resolver.go]: technically a policy can have the same peer in source and dest but would not make any sense as a peer can not be linked to itself.

One thing that we ar emissing is that a policy is not always group <-> group and peer <-> peer but can also be a mix of peer <-> group or group <-> peer. This needs to reflect in the lookup

Comment (dmitri-netbird) [reply management/server/affectedpeers/resolver.go]: done
